Lib/fontTools/ttLib/tables/TupleVariation.py

```diff
@@ -446,6 +444,118 @@ def getTupleSize_(flags, axisCount):
 			size += axisCount * 4
 		return size
 
+	def getCoordWidth(self):
+		""" Return 2 if coordinates are (x, y) as in gvar, 1 if single values
+		as in cvar, or 0 if empty.
+		"""
+		firstDelta = next((c for c in self.coordinates if c is not None), None)
+		if firstDelta is None:
+			return 0  # empty or has no impact
+		if type(firstDelta) is tuple and len(firstDelta) == 2:
+			return 2
+		elif type(firstDelta) in (int, float):
+			return 1
+		else:
+			raise TypeError("invalid type of delta: %s" % type(firstDelta))
+
+	def scaleDeltas(self, scalar):
+		if scalar == 1.0:
+			return  # no change
+		coordWidth = self.getCoordWidth()
+		if coordWidth == 2:
+			self.coordinates = [
+				(d[0] * scalar, d[1] * scalar) if d is not None else None
+				for d in self.coordinates
+			]
+		elif coordWidth == 1:
+			self.coordinates = [
+				d * scalar if d is not None else None
+				for d in self.coordinates
+			]
```

Comment:
I understand Rod's comments about the coding being duplicated between the coordWidth == 1 and coordWidth == 2 cases. It's annoying that both the loop over self.coordindates and the check for None are duplicated.

You mentioned below that there are some semantical differences regarding None in some situations, but i'm not seeing any in this function or in roundDeltas below.  

The biggest problem is that coordWidth == 2 is operating on tuples while coordWidth == 1 is operating on scalars.

I did come up with this:
   self.coordinates = [None if d is None else d * scalar if coordWidth == 1 else (d[0] * scalar, d[1] * scalar) for d in deltas]

This has the loop and the check for None once rather than twice.

Reply:
ok, I chained the two if statements in one line like you proposed.